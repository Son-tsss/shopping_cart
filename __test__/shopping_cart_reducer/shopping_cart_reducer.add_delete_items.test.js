import shoppingCartReducer from "../../src/shopping_cart_reducer"

it('should add new item to the items list', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            }
        ],
        discount: 1
    }

    const newItem = {
        id: "i2",
        price: 200
    }

    const real = shoppingCartReducer(prevState, {type: "ADD_ITEM", ...newItem});

    expect(real).toMatchSnapshot()
});

it('should delete existing item from items list', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 1
            }
        ],
        discount: 1
    }

    const real = shoppingCartReducer(prevState, {type: "DELETE_ITEM", id: "i2"});

    expect(real).toMatchSnapshot();
});

it('deleting not existing item from items list should changes nothing', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 1
            }
        ],
        discount: 1
    }

    const real = shoppingCartReducer(prevState, {type: "DELETE_ITEM", id: "i3"});

    expect(real).toMatchSnapshot();
});
