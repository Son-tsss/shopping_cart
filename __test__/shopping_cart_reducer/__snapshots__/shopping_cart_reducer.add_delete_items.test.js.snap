// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleting not existing item from items list should changes nothing 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 2,
      "discount": 0.5,
      "id": "i1",
      "price": 100,
    },
    Object {
      "count": 1,
      "discount": 1,
      "id": "i2",
      "price": 200,
    },
  ],
}
`;

exports[`should add new item to the items list 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 2,
      "discount": 0.5,
      "id": "i1",
      "price": 100,
    },
    Object {
      "count": 1,
      "discount": 1,
      "id": "i2",
      "price": 200,
    },
  ],
}
`;

exports[`should delete existing item from items list 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 2,
      "discount": 0.5,
      "id": "i1",
      "price": 100,
    },
  ],
}
`;
